packages/schema/src/config/common.ts

```diff
@@ -156,9 +156,7 @@ export default defineUntypedSchema({
    */
   serverDir: {
     $resolve: async (val: string | undefined, get): Promise<string> => {
-      const isV4 = ((await get('future') as Record<string, unknown>).compatibilityVersion === 4)
-
-      return resolve(isV4 ? await get('rootDir') as string : await get('srcDir') as string, val ?? 'server')
+      return resolve(await get('rootDir') as string, val ?? 'server')
```

Comment:
A little tweak required.

if it's not v4 compat, then by default this should be `<srcDir>/server`. In v4 compat, it should be `<rootDir>/server`. If the user provides it then in either case it should be resolved relative to `<rootDir>`.